plugins/@grouparoo/calculated-property/src/lib/profileProperty.ts

```
@@ -0,0 +1,72 @@
+import {
+  GrouparooCLI,
+  plugin,
+  Profile,
+  ProfilePropertyPluginMethod,
+} from "@grouparoo/core";
+import { NodeVM } from "vm2";
+
+async function calculateProfilePropertyValue(
+  customFunction,
+  profile: Profile,
+  propertyName: string
+): Promise<string> {
+  let populatedFunction;
+  try {
+    populatedFunction = await plugin.replaceTemplateProfileVariables(
+      customFunction,
+      profile,
+      false
+    );
+    //fail at every level if someone tries to require a library... this should never be allowed to hit vm.run
```

This means that running `roo validate` or `roo apply` will fail.  It fails a bit more abruptly than with other config errors, but I think that's warranted.  We don't want anything questionable attempting to be executed.

---

I think this can be tested! Wound't this throw in the way you are already testing in `__tests__/import/import-property.ts ` ?

---

To confirm - this check is to "fail fast" with `grouparoo validate` if you try to require something or read process.env.  Otherwise, the function would be valid, but produce results you don't expect as we block / null out those at runtime.



We can test that `process.env` is empty with a test like 

```js
const fn = `() => { return process.env.DATABASE_URL }`; 
```

and show that it's `undefined` which might be a good idea.

---

It wasn't throwing the exact same way, but I've changed it to now